### PR TITLE
Implement locksmith for CoreOs upgrades (fixes #812).

### LINKED ISSUE #812: Implement rebootStrategy for CoreOS
We do not appear to reference `osConfig.rebootStrategy` anywhere in kraken.

```
rodin:k2 dwat$ grep -r rebootStrategy ansible
ansible/roles/kraken.config/files/config-private-topology.yaml:      rebootStrategy: "off"
ansible/roles/kraken.config/files/config.yaml:      rebootStrategy: "off"
rodin:k2 dwat$ grep -r rebootStrategy ansible
```

As a result, we get the default reboot strategy, which is not `off` like it is in kraken config.

```
core@ip-10-0-125-53 ~ $ journalctl -u locksmithd
-- Logs begin at Fri 2017-09-08 15:43:20 UTC, end at Fri 2017-09-08 21:09:40 UTC. --
Sep 08 15:43:33 localhost systemd[1]: Started Cluster reboot manager.
Sep 08 15:43:33 ip-10-0-125-53.ec2.internal locksmithd[712]: Reboot strategy "best-effort" is deprecated and will be removed in the future.
Sep 08 15:43:33 ip-10-0-125-53.ec2.internal locksmithd[712]: Please explicitly set the reboot strategy to one of [off reboot etcd-lock]
Sep 08 15:43:33 ip-10-0-125-53.ec2.internal locksmithd[712]: See https://coreos.com/os/docs/latest/update-strategies.html for details on configuring reboot strategies.
Sep 08 15:43:33 ip-10-0-125-53.ec2.internal locksmithd[712]: No configured reboot window
Sep 08 15:43:33 ip-10-0-125-53.ec2.internal locksmithd[712]: locksmithd starting currentOperation="UPDATE_STATUS_IDLE" strategy="best-effort"
Sep 08 15:43:38 ip-10-0-125-53.ec2.internal locksmithd[712]: [etcd.service etcd2.service] are inactive
Sep 08 15:43:38 ip-10-0-125-53.ec2.internal locksmithd[712]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 10s.
Sep 08 15:43:48 ip-10-0-125-53.ec2.internal locksmithd[712]: [etcd.service etcd2.service] are inactive
Sep 08 15:43:48 ip-10-0-125-53.ec2.internal locksmithd[712]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 20s.
Sep 08 15:43:59 ip-10-0-125-53.ec2.internal systemd[1]: Stopping Cluster reboot manager...
Sep 08 15:43:59 ip-10-0-125-53.ec2.internal systemd[1]: Stopped Cluster reboot manager.
Sep 08 15:43:59 ip-10-0-125-53.ec2.internal systemd[1]: Started Cluster reboot manager.
Sep 08 15:43:59 ip-10-0-125-53.ec2.internal locksmithd[1121]: No configured reboot window
Sep 08 15:43:59 ip-10-0-125-53.ec2.internal locksmithd[1121]: locksmithd starting currentOperation="UPDATE_STATUS_IDLE" strategy="etcd-lock"
Sep 08 15:46:56 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=0 Progress=0 CurrentOperation="UPDATE_STATUS_CHECKING_FOR_UPDATE" NewVersion=0.0.0 NewSize=0
Sep 08 15:46:56 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504885616 Progress=0 CurrentOperation="UPDATE_STATUS_IDLE" NewVersion=0.0.0 NewSize=0
Sep 08 16:32:25 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504885616 Progress=0 CurrentOperation="UPDATE_STATUS_CHECKING_FOR_UPDATE" NewVersion=0.0.0 NewSize=0
Sep 08 16:32:25 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504888345 Progress=0 CurrentOperation="UPDATE_STATUS_IDLE" NewVersion=0.0.0 NewSize=0
Sep 08 17:18:28 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504888345 Progress=0 CurrentOperation="UPDATE_STATUS_CHECKING_FOR_UPDATE" NewVersion=0.0.0 NewSize=0
Sep 08 17:18:28 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504891108 Progress=0 CurrentOperation="UPDATE_STATUS_IDLE" NewVersion=0.0.0 NewSize=0
Sep 08 18:07:19 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504891108 Progress=0 CurrentOperation="UPDATE_STATUS_CHECKING_FOR_UPDATE" NewVersion=0.0.0 NewSize=0
Sep 08 18:07:19 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504894039 Progress=0 CurrentOperation="UPDATE_STATUS_IDLE" NewVersion=0.0.0 NewSize=0
Sep 08 18:57:07 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504894039 Progress=0 CurrentOperation="UPDATE_STATUS_CHECKING_FOR_UPDATE" NewVersion=0.0.0 NewSize=0
Sep 08 18:57:07 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504897027 Progress=0 CurrentOperation="UPDATE_STATUS_IDLE" NewVersion=0.0.0 NewSize=0
Sep 08 19:39:37 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504897027 Progress=0 CurrentOperation="UPDATE_STATUS_CHECKING_FOR_UPDATE" NewVersion=0.0.0 NewSize=0
Sep 08 19:39:37 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504899577 Progress=0 CurrentOperation="UPDATE_STATUS_IDLE" NewVersion=0.0.0 NewSize=0
Sep 08 20:22:32 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504899577 Progress=0 CurrentOperation="UPDATE_STATUS_CHECKING_FOR_UPDATE" NewVersion=0.0.0 NewSize=0
Sep 08 20:22:32 ip-10-0-125-53.ec2.internal locksmithd[1121]: LastCheckedTime=1504902152 Progress=0 CurrentOperation="UPDATE_STATUS_IDLE" NewVersion=0.0.0 NewSize=0
core@ip-10-0-125-53 ~ $ 
```

### PR DESCRIPTION
**Pull request by davidewatson**

### CODE REVIEW

File: ansible/roles/kraken.cluster_common/templates/os.update.part.jinja2 [hunk 1]
```diff
@@ -1,4 +1,2 @@
-group: {{item.osConfig.channel}}
-{% if item.bastionConfig is not defined %}
```
Comment (davidewatson) [inline ansible/roles/kraken.cluster_common/templates/os.update.part.jinja2]: @alejandroEsc: Why is this here? It makes it more difficult to understand the os upgrade behavior of bastion nodes. If they want a different os upgrade behavior, it would be more clear if they had a different osConfig.

Comment (davidewatson) [reply ansible/roles/kraken.cluster_common/templates/os.update.part.jinja2]: @samba, @notjames: ^^ It would be nice if the default CoreOS upgrade strategy used by kraken were the same as that which is recommended and used by the Cluster Ops team. This PR will make `etcd-lock` the default for all nodePools. This can be changed on a per nodePool basis.

**Comment by davidewatson** [comment]
@samba, @notjames: ^^ It would be nice if the default CoreOS upgrade strategy used by kraken were the same as that which is recommended and used by the Cluster Ops team. This PR will make `etcd-lock` the default for all nodePools. This can be changed on a per nodePool basis.

**Comment by davidewatson** [comment]
I have changed the default behavior to `off` instead of `etcd-lock`. This is necessary for reliable CI. Cluster Ops and Customer teams will need to adjust it accordingly.

File: ansible/roles/kraken.cluster_common/templates/locksmith.part.jinja2 [hunk 2]
```diff
@@ -1,4 +1,31 @@
...
+etcd_cafile: /etc/{{ etcdConfig.name }}/ssl/client-ca.pem
+etcd_certfile: /etc/{{ etcdConfig.name }}/ssl/client-certfile.pem
+etcd_keyfile: /etc/etcd/{{ etcdConfig.name }}/client-keyfile.pem
+{% else %}
+endpoint: {{ locksmith.endpoint }}
+etcd_cafile: {{ locksmith.etcdCafile }}
+etcd_certfile: {{ locksmith.etcdCertfile }}
+etcd_keyfile: {{ locksmith.etcdKeyfile }}
+{% endif %}{# locksmith.etcdConfig is defined #}
+{% endif %}{# if item.osConfig.locksmith is defined #}
```
Comment (joejulian) [inline ansible/roles/kraken.cluster_common/templates/locksmith.part.jinja2]: Is this similar to what you called @coffeepac out on? Logic in the template?

Comment (davidewatson) [reply ansible/roles/kraken.cluster_common/templates/locksmith.part.jinja2]: Kind of. Here we are just inserting config values into the template. We are not considering whether a particular combination is valid or not since all the syntactic and semantic validation should have already happened inside of Ansible. Point taken though.

**Comment by davidewatson** [comment]
k2/815 fixes kraken so that it respects the `rebootStrategy` option in the config. The config specified `off`, meaning no automatic reboot would occur if an CoreOS update was staged. Prior to this fix, `best-effort`, which is the CoreOS default, was aways used instead. Note that `best-effort` is being depreciated on Sept 14. See https://coreos.com/blog/locksmith-update-strategy-revision for more information.
